File: gaia-sdk-javascript/src/graphql/request/input/UpdateTenantImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to update a tenant
 */
export class UpdateTenantImpulse {
public _typeName = "UpdateTenantImpulse";
    private tenantId:Uuid;
    private qualifier:String;

    constructor (tenantId:Uuid, qualifier:String) {
        this.tenantId = tenantId;
        this.qualifier = qualifier;
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/request/input/UpdateTenantImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class UpdateTenantImpulse():
    """
    The specification to update a tenant
    """
    tenantId: str
    qualifier: str
    implicitIdentities: List[str]
    explicitIdentities: List[str]

    def __init__(self, tenantId: str, qualifier: str, implicitIdentities: List[str], explicitIdentities: List[str]):
        self.tenantId = tenantId
        self.qualifier = qualifier
        self.implicitIdentities = implicitIdentities
        self.explicitIdentities = explicitIdentities

    def __eq__(self, other):
        if type(other) is type(self):
            return self.tenantId == other.tenantId and self.qualifier == other.qualifier and self.implicitIdentities == other.implicitIdentities and self.explicitIdentities == other.explicitIdentities
        return False

    def __repr__(self):
        return {'tenantId': self.tenantId, 'qualifier': self.qualifier, 'implicitIdentities': self.implicitIdentities, 'explicitIdentities': self.explicitIdentities}
